source/s3_auto_ranged_put.c

```diff
@@ -332,88 +271,44 @@ static int s_s3_auto_ranged_put_next_request(
 static int s_s3_auto_ranged_put_prepare_request(
     struct aws_s3_meta_request *meta_request,
     struct aws_s3_client *client,
-    struct aws_s3_request *request) {
+    struct aws_s3_vip_connection *vip_connection,
+    bool is_initial_prepare) {
     AWS_PRECONDITION(meta_request);
     AWS_PRECONDITION(client);
+    (void)client;
+    (void)is_initial_prepare;
+
+    struct aws_s3_request *request = vip_connection->request;
     AWS_PRECONDITION(request);
 
     struct aws_s3_auto_ranged_put *auto_ranged_put = meta_request->impl;
     AWS_PRECONDITION(auto_ranged_put);
 
     struct aws_http_message *message = NULL;
-    struct aws_s3_part_buffer *part_buffer = NULL;
-
-    /* If we're not sending the entire object, then we need to grab a part buffer */
-    if (request->desc_data.request_tag != AWS_S3_AUTO_RANGED_PUT_REQUEST_TAG_ENTIRE_OBJECT) {
```

Comment:
If we need to upload the entire file in one go, then we currently use the default meta request.